src/Components/Form/index.jsx

```diff
@@ -1,34 +1,42 @@
-import React from 'react';
+import { useState } from 'react';
 import './Form.scss';
 
-function Form (props) {
+function Form(props) {
+  const [url, setUrl] = useState('');
+  const [method, setMethod] = useState('GET');
+  const [postData, setPostData] = useState('');
 
-  let handleSubmit = e => {
+
+  const handleSubmit = (e) => {
     e.preventDefault();
     const formData = {
-      method:'GET',
-      url: 'https://pokeapi.co/api/v2/pokemon',
+      method: method,
+      url: url,
+      data: postData,
     };
     props.handleApiCall(formData);
-  }
+  };
+
 
-    return (
-      <>
-        <form onSubmit={handleSubmit}>
-          <label >
-            <span className='url-input-bar'>URL: </span>
-            <input name='url' type='text' />
-            <button type="submit">GO!</button>
-          </label>
-          <label className="methods">
-            <span id="get">GET</span>
-            <span id="post">POST</span>
-            <span id="put">PUT</span>
-            <span id="delete">DELETE</span>
-          </label>
-        </form>
-      </>
-    );
+  return (
+    <>
+      <form onSubmit={handleSubmit}>
+        <label >
+          <span className='url-input-bar'>URL: </span>
+          <input data-testid="form-input" name='url' type='text' onChange={(e) => setUrl (e.target.value)}/>
+          <button data-testid="form-button" type="submit">GO!</button>
+        </label>
+        <label className="methods">
+          <span data-testid="form-span-get" id="get" onClick={(e) => setMethod('GET')}>GET</span>
+          <span data-testid="form-span-post" id="post" onClick={(e) => setMethod('POST')}>POST</span>
+          <span data-testid="form-span-put" id="put" onClick={(e) => setMethod('PUT')}>PUT</span>
+          <span data-testid="form-span-delete" id="delete" onClick={(e) => setMethod('DELETE')}>DELETE</span>
+        </label>
+        {method === 'POST' && <textarea onChange={(e) => setPostData(e.target.value)}/>}
+        {method === 'PUT' && <textarea onChange={(e) => setPostData(e.target.value)}/>}
```

Comment:
You can combine these into a single line to avoid repeating code by chaining your logical operators. Something like:
```javascript
{(method === 'POST' || method === 'PUT') && <textarea onChange={(e) => setPostData(e.target.value)}/>}
```